src/services/auth-api/index.ts

```diff
@@ -20,28 +21,42 @@ class AuthApi {
     this.#apiPath = apiPath;
   }
 
-  public signIn(payload: ISignInPayload): Promise<AuthResponse> {
-    return this.#http.load(`${this.#apiPath}${ApiPath.SIGN_IN}`, {
-      method: HttpMethod.POST,
-      contentType: ContentType.JSON,
-      payload: JSON.stringify(payload),
-      hasAuth: false,
-    });
+  public async signIn(
+    payload: ISignInPayload
+  ): Promise<AuthResponse | undefined> {
+    try {
+      return this.#http.load(`${this.#apiPath}${ApiPath.SIGN_IN}`, {
+        method: HttpMethod.POST,
+        contentType: ContentType.JSON,
+        payload: JSON.stringify(payload),
+        hasAuth: false,
+      });
+    } catch (err: any) {
+      notify(err?.message ?? 'Failed to log in');
```

Comment:
Also we need to rename it probably, to make it more clear. "Notify" could mean a lot, let's try something like 'Toaster' or similar, let's choose.